modules/Classroom/components/StudentHeader/StudentHeader.tsx

```diff
@@ -1,24 +1,40 @@
+import { useState } from "react";
+
 import { ActionIcon, Flex, Title } from "@mantine/core";
 import { FaRegPlusSquare } from "react-icons/fa";
 
 import { useAppSelector } from "@/shared/redux/hooks";
 import { selectAuthenticatedUser } from "@/shared/redux/reducers/user.reducer";
 import { EUserRole } from "@/shared/redux/rtk-apis/auth/auth.types";
 
-const StudentHeader: React.FC = () => {
+import StudentSearchModal from "../StudentSearchModal/StudentSearchModal";
+import { StudentHeaderProps } from "./StudentHeader.interface";
+
+const StudentHeader: React.FC<StudentHeaderProps> = ({ classroomId }) => {
   const user = useAppSelector(selectAuthenticatedUser);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const handleOpenModal = () => setIsModalOpen(true);
+  const handleCloseModal = () => setIsModalOpen(false);
```

Comment:
[NIT] - These are too simple for separate methods in my opinion. I would inline it in the prop